### PR TITLE
Add tags to instance in CloudStack builder

### PR DESCRIPTION
**Pull request by aleksey-hariton**
Add tags to instance when creating

### CODE REVIEW

**Comment by aleksey-hariton** [comment]
Guys, failed tests not related to my changes =/

**Comment by SwampDragons** [comment]
We've been having a lot of flakiness with Travis lately. I'll rerun.

**Comment by aleksey-hariton** [comment]
Config example:

```json
"builders": [
    {
      "type": "cloudstack",
       
       ...

      "tags": {
        "Environment": "dev",
        "Product": "packer"
      },
...
]
```

File: builder/cloudstack/step_create_instance.go [hunk 1]
```diff
@@ -149,6 +149,20 @@ func (s *stepCreateInstance) Run(ctx context.Context, state multistep.StateBag)
 	// Store the instance ID so we can remove it later.
 	state.Put("instance_id", instance.Id)
 
+	// Set instance tags
+	if config.Tags != nil {
```
Comment (SwampDragons) [inline builder/cloudstack/step_create_instance.go]: When not set, is config.Tags definitely a nil map, or is it just an empty map?  I can imagine this check passing if tags is empty; example: https://play.golang.org/p/EA8lq7Bpdyh

**Comment by SwampDragons** [comment]
Thanks for your patience in getting a review; I don't have a cloudstack setup so I was hoping one of our more cloudstack-savvy folks could review this. It looks pretty straightforward, though, and you've waited long enough. I have just one question inline and if you let me know your thoughts we can get this merged.

File: builder/cloudstack/step_create_instance.go (hunk 1 above)
Comment (azr) [reply builder/cloudstack/step_create_instance.go]: One alternative could be to use len() 🙂

**Pull request merged**